autoapi/mappers/base.py

```diff
@@ -176,14 +210,19 @@ def create_class(self, obj):
     def output_rst(self, root, source_suffix):
         for id, obj in self.objects.items():
 
-            if not obj:
+            if not obj or not obj.top_level_object:
                 continue
 
             rst = obj.render()
             if not rst:
                 continue
 
-            detail_dir = os.path.join(root, *id.split('.'))
+            try:
+                filename = id.split('(')[0]
+            except IndexError:
+                filename = id
+            filename = filename.replace('#', '-')
```

Comment:
What's this for? Is this .NET `Foobar#ctor` parsing? Are there more cases to be slugged here, or does it make sense to move this to the .NET mapper?